### ISSUE TITLE
Test Npcap (and WinPcap and Nmap) on Windows 11 Preview Build

### ISSUE THREAD

**Issue by fyodor** [nmap/npcap#521]
Microsoft is planning to release Windows 11 in late 2021 and they already have a [Insider Preview Version](https://blogs.windows.com/windows-insider/2021/06/28/announcing-the-first-insider-preview-for-windows-11/) available for folks signed up to the Windows Insider program.  Once Windows 11 is released, adoption is expected to be particularly fast since they are providing it as a free update using Windows Update to Windows 10 users.  The changes don't seem particularly dramatic, but who knows. It would be great if we can identify any potential problems in advance so we don't have to frantically scramble once the release starts rolling out to the masses.

While we want to test our own software (Npcap and Nmap) for obvious reasons, we'd also like to know whether WinPcap continues to (partially) work on Windows 11 like it works (somewhat) on Windows 10.  Because if it doesn't work on Windows 11, that will likely ignite a mad scramble by whatever WinPcap users remain toward Npcap.  So we should prepare for that.

If anyone has tested Npcap or the old WinPcap or Nmap on the Windows 11 platform on x86 or ARM platforms, please give a report here.  Even if it all works fine, that is helpful to know. Thanks!

**Comment by njk42** [comment]
I am developing a 1G Ethernet-based UDP/broadcast streaming application that uses pcap to sniff packets on the desired network adapter/port and send them to my application.  (The eventual application is similar to GigE-Vision, but is not using that standard in particular).

With Windows **10** and npcap (v1.55), I can reliably transfer >800 Mbps and have no packet loss on a variety of consumer-grade PC platforms (laptops, desktops, single-board computers, etc) available over the last~5 years  (ie. so nothing special or high-horsepower required from a hardware perspective to get 100% of packets captured). 

Now I just upgraded to a shiny new MS Surface Pro 8 tablet (which requires Windows **11**, build 22000) and installed my test software and Npcap expecting similar performance to my Windows 10 experience, but unfortunately the packet loss figures are intolerable (i.e. steady packet loss about 1-2%) even with the data generation dialed down to ~200 Mbps. 

I don't know if this is an issue/incompatibility with Npcap and Windows 11, or an inherent limitation (bug!) in Windows 11 itself, nor can I even confirm whether or not it's related to this specific Surface Pro 8 hardware (since MS won't let me _downgrade_ to Windows 10 and compare apples-to-apples).

So anyway, it's not a conclusive or scientific data point, but I thought it would be worth mentioning here in case someone else has seen similar behavior on Windows 11 and knows of any fixes (change to settings, versions, etc).  FWIW, I disabled the firewall, file sharing, and other services on Windows 11 and it showed no improvement. 

Sometime in the future, I will find another PC which can be updated from Windows 10 to Windows 11 and measure packet performance before and after OS update.  This could help remove any doubts about hardware-related deficiencies and put the issue squarely on Win11, but generally I already think the likelihood of this being a hardware limitation of the Surface Pro 8 is pretty low... it has a modern i7 processor, 16 GB RAM, yada yada and it otherwise a very snappy system.

**Comment by fyodor** [comment]
Thanks for the data point, @njk42.  It's unfortunate that performance was so low on that Windows 11 Surface Pro 8 tablet.  When you do that future testing you mentioned (such as Win10 vs Win11 on same hardware), please do post the results.  We're also happy to hear about the experiences of anyone else using Npcap with Windows 11.

**Comment by njk42** [comment]
UPDATE:  I _down_graded the Surface Pro 8 tablet to Windows 10 _Home_ (build 19043) despite concerns that I may run into missing drivers for the Surface hardware pre-installed with Windows 11.  Luckily all devices in Device Manager seem to have installed drivers.

Retesting the high-bandwidth UDP stream shows no improvement in packet loss however, so it appears my earlier report may not be related to a Windows 11 issue but rather either a hardware issue with this tablet, or some other OS difference between the Home edition of Win10 and the Pro edition (on which all of my previous successful testing has been performed).  Although I would not expect the network driver stack to be different in Home vs Pro, I guess anything is possible with Microsoft and I will likely update this PC to the $99 Pro edition for further testing.

**Comment by njk42** [comment]
UPDATE #2:  I found an older Surface Pro 6 which had Win 10 Home;  tested in the same fashion as with the Surface Pro 8, and UDP packet stream works well.  I then upgraded the Surface Pro 6 to Win 11 Home and packet stream still works well!   Npcap version 1.55 was used.   

So it appears my inability to get good packet delivery performance (as mentioned in my earlier posts last week) is specific to the Surface Pro 8 hardware (or it's specific HW drivers).  From my perspective, Win 11 and Npcap are working fine together on the Surface Pro 6 hardware.
